src/manage_sql_filters.c

```diff
@@ -1188,7 +1188,13 @@ filter_clause (const char* type, const char* filter,
                 g_string_append_printf
                  (order,
                   " ORDER BY"
-                  "  (CASE WHEN (SELECT target = 0 FROM tasks"
+                  "  (CASE WHEN (SELECT "
+                  "               ( "
+                  "                  target = 0 "
+                  "                  and COALESCE( agent_group, 0) = 0 "
+                  "                  and COALESCE( oci_image_target, 0) = 0"
+                  "               ) "
+                  "              FROM tasks"
                   "              WHERE tasks.id = task)"
                   "    THEN 'Container'"
```

Comment:
Here it's using "Container" to make sure the container tasks are sorted first. Do you think it might confuse people? Maybe it could be:
```suggestion
                  "    THEN '0'" // Sort first.
```